docs/dg/dev/third-party-module-development/ensure-quality.md

```diff
@@ -0,0 +1,98 @@
+
+## Tools
+
+To make sure that code quality matched the Spryker standards it is important to execute code quality tools, here is the list:
+
+| Command                                                                                 | Description                                                             |
+|:----------------------------------------------------------------------------------------|:------------------------------------------------------------------------| 
+| console code:propel:validate-abstract -m YourCompanyName.YourModuleName                 | Validates that propel entity/query classes are implemented properly.    |
+| console propel:schema:validate                                                          | Validates Propel the schema files.                                      |
+| console propel:schema:validate-xml-names                                                | Validates Propel XML element name rules for schema files.               |
+| console transfer:validate                                                               | Validates transfer XML definition files.                                |
+| spryker-dev-console dev:validate-module-schemas YourCompanyName.YourModuleName          | Validates Propel schema definitions.                                    |
+| spryker-dev-console dev:validate-module-databuilders YourCompanyName.YourModuleName     | Validates databuilder XML files.                                        |
+| spryker-dev-console dev:module-meta-files -d -v YourCompanyName.YourModuleName          | Validates that all the mandatory meta files exist.                      |
+| spryker-dev-console dev:composer:update-json-files -d -v YourCompanyName.YourModuleName | Validates that all the necessary module dependencies are added.         |
+| console dev:composer:validate-json-files YourCompanyName.YourModuleName                 | Validates composer.json file to make sure that it's structure is valid. |
+| vendor/your-company-name/your-module-name/console code:sniff:style                      | Sniffs and fixes code style.                                            |
+| spryker-dev-console dev:validate-module-transfers -m YourCompanyName.YourModuleName     | Validates DTO definition for the module.                                |
+| vendor/your-company-name/your-module-name/console code:sniff:architecture               | Validates module architecture to find a common architecture mistakes.   |
+| vendor/your-company-name/your-module-name/console code:phpstan                          | Runs PHPStan static analyzer to ensure the code quality.                |
+| vendor/your-company-name/your-module-name/console code:phpmd                            | Runs PHPMD  to detect code smells and possible errors within the analyzed source code.                |
+
+## GitHub CI file example 
+
+Make sure that your module CI runs all the command above
```

Comment:
My previous comment has disappeared somehow. So I will once again mention that these commands imply the need for external dependencies. Where are these dependencies mentioned?

Reply:
Updated list of commands to have only those that work in standalone mode, we'll extend this list in the future when finish the development of related functionality.